component/alertrules.libsonnet

```
@@ -16,63 +14,26 @@ assert
   std.member(inv.applications, 'openshift4-monitoring')
   : 'Component `openshift4-monitoring` not enabled';
 
-// Keep config backwards compatible
-local predict_storage_alert = elasticsearch.predict_elasticsearch_storage_alert + (
-  if std.objectHas(params, 'predict_elasticsearch_storage_alert') then
-    std.trace(
-      'parameter predict_elasticsearch_storage_alert is deprecated, please use parameter `components.elasticsearch.predict_elasticsearch_storage_alert instead`',
-      com.makeMergeable(params.predict_elasticsearch_storage_alert)
-    )
-  else {}
-);
-
+// Upstream alerts to ignore
 // Keep only alerts from params.ignore_alerts for which the last
 // array entry wasn't prefixed with `~`.
-local user_ignore_alerts = com.renderArray(params.ignore_alerts);
-
-// Upstream alerts to ignore
 local ignore_alerts = std.set(
   // Add set of upstream alerts that should be ignored from processed value of
   // `params.ignore_alerts`
-  user_ignore_alerts
+  com.renderArray(std.get(params, 'ignore_alerts', []))
+  + com.renderArray(std.get(params, 'ignore_alerts', []))
```

One of these should be `alerts.ignore` I assume? Additionally, I'd add a deprecation warning for the deprecated field.

```suggestion
  local old_ignore = if std.objectHas(params, 'ignore_alerts') then
    std.trace('Parameter `ignore_alerts` is deprecated, please migrate your config to `alerts.ignore`.', params.ignore_alerts)
  else
    []; 
  com.renderArray(params.alerts.ignore)
  + com.renderArray(old_ignore)
```

---

yes, right 👍